app/scripts/modules/core/cluster/categorySelection/categorySelection.service.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+let angular = require('angular');
+
+module.exports = angular.module('spinnaker.categorySelection.service', [
+  require('../../account/account.service.js'),
+  require('../../config/settings.js'),
+  require('../../utils/lodash.js'),
+])
+  .factory('categorySelectionService', function($uibModal, $q, _, accountService, settings) {
+    function availableCategories() {
+      var available = [
+        {
+          name: 'serverGroup',
+          display: 'Server Group',
+        },
+      ];
+
+      if (settings.feature.jobs) {
+        available = available.concat([
+          {
+            name: 'job',
+            display: 'Job',
+          },
+        ]);
+      }
+
+      return available;
+    }
+
+    function selectCategory() {
+      return $q.when((() => {
+        var categories = availableCategories();
+        var category;
+
+        if (categories.length > 1) {
+          category = $uibModal.open({
+            templateUrl: require('./categorySelection.html'),
+            controller: 'CategorySelectCtrl as ctrl',
+            resolve: {
+              categoryOptions: function() { return categories; }
+            }
+          }).result;
+        } else if (categories.length === 1) {
+          category = $q.when(categories[0].name);
+        } else {
+          category = $q.when(settings.defaultCategory || 'serverGroup');
+        }
+        return category;
+      })());
```

Comment:
Why the IIFE? Does this need to be wrapped and immediately invoked for some reason?